src/main/java/com/sypht/CloseableHttpClient.java

```diff
@@ -17,10 +18,59 @@ public CloseableHttpClient() {
 
         HttpClientConnectionManager poolingConnManager
                 = new PoolingHttpClientConnectionManager();
-        this.httpClient = HttpClients
-                .custom()
-                .setConnectionManager(poolingConnManager)
-                .build();
+
+        HttpHost proxy = getSystemProxy();
+        if (proxy!=null) {
+            this.httpClient = HttpClients
+                    .custom().setProxy(proxy)
+                    .setConnectionManager(poolingConnManager)
+                    .build();
+        } else {
+            this.httpClient = HttpClients
+                    .custom()
+                    .setConnectionManager(poolingConnManager)
+                    .build();
+        }
+    }
+
+    private HttpHost getSystemProxy() {
+        HttpHost proxy = null;
+        String proxyHost = null;
+        Integer proxyPort = null;
+        String httpProxyHost = System.getProperty("http.proxyHost");
+        String httpsProxyHost = System.getProperty("https.proxyHost");
+        String socksProxyHost = System.getProperty("socksProxyHost");
+
+        if (httpProxyHost != null) {
+            proxyHost = httpProxyHost;
+            String httpProxyPortStr = System.getProperty("http.proxyPort");
```

Comment:
you may also wrap System.getProperty() in a method that can tries both System.getenv() and System.getProperty() in sequence. Most modern runtimes such as docker support only env.